### PR TITLE
chore: add UserService and Implementation for register

### PR DESCRIPTION
**Pull request by ramesz6**

### CODE REVIEW

File: backend/src/main/java/com/greenfoxacademy/backend/services/UserServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+import com.greenfoxacademy.backend.models.User;
+import com.greenfoxacademy.backend.repositories.UserRepository;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserServiceImpl implements UserService {
+    UserRepository userRepository;
+
+
```
Comment (markkovari) [inline backend/src/main/java/com/greenfoxacademy/backend/services/UserServiceImpl.java:11]: Why two newlines here?

File: backend/src/main/java/com/greenfoxacademy/backend/services/UserService.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.greenfoxacademy.backend.services;
+
+import com.greenfoxacademy.backend.models.User;
+import org.springframework.stereotype.Service;
+
+@Service
+public interface UserService {
+    public void register(User newUser);
+
```
Comment (markkovari) [inline backend/src/main/java/com/greenfoxacademy/backend/services/UserService.java:9]: Why this newline here?

File: backend/src/main/java/com/greenfoxacademy/backend/services/UserServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package com.greenfoxacademy.backend.services;
+
+import com.greenfoxacademy.backend.models.User;
+import com.greenfoxacademy.backend.repositories.UserRepository;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserServiceImpl implements UserService {
+    UserRepository userRepository;
```
Comment (markkovari) [inline backend/src/main/java/com/greenfoxacademy/backend/services/UserServiceImpl.java:9]: Shouldn't this be private and maybe final too?
When is is going to be created, should't we go with dependency inhjection within constructors?

**Review by markkovari — COMMENTED** [review]
Just a few nitpicks that even the reviewdog did not pick up. We need to train him/her/they better.

**Review by carlos-ponton — APPROVED** [review]

**Review by AnnaUgrai — APPROVED** [review]
https://i.imgflip.com/46pfbw.jpg

**Pull request merged**